### PR TITLE
Fallback to iojs

### LINKED ISSUE #8987: Node Tool Installer fails on versions [0-3].x
## Environment
- Server - Azure Pipelines
- Agent - Hosted VS2017

## Issue Description

The node tool installer fails with a 404 response.

image

Upon inspection, it looks like it is expecting there to be packages containing node for windows at https://nodejs.org/dist/v0.12.18 that aren't there. This isn't an issue for other newer versions (e.g. https://nodejs.org/dist/v6.13.1/node-v6.13.1-win-x64.7z exists).

### Task logs

More verbose logs:
image

zip: [log_21_12.zip](https://github.com/Microsoft/azure-pipelines-tasks/files/2640225/log_21_12.zip)

### PR DESCRIPTION
**Pull request by damccorm**
Right now, we fail to find node if the version specified is 1.x, 2.x, or 3.x. This is because nodejs was forked for v1-3 into iojs and merged for v4. Right now, node basically considers iojs to be node 1-3 so this PR looks for iojs versions if node isn't available.

Fixes #8987

### CODE REVIEW

**Comment by stephenmichaelf** [comment]
Could you add some context on what this is for? Thanks!

**Comment by damccorm** [comment]
@stephenmichaelf sorry, should have had that originally. Updated the initial comment to provide context.

File: Tasks/NodeToolV0/nodetool.ts [hunk 1]
```diff
@@ -63,7 +64,12 @@ async function getNode(versionSpec: string, checkLatest: boolean) {
             // query nodejs.org for a matching version
             version = await queryLatestMatch(versionSpec);
             if (!version) {
-                throw new Error(`Unable to find Node version '${versionSpec}' for platform ${osPlat} and architecture ${osArch}.`);
+                // If we can't find it from nodejs.org, try iojs.org (has versions 1-3)
```
Comment (stephenmichaelf) [inline Tasks/NodeToolV0/nodetool.ts]: Add an extra condition to also check the version we are looking for and make sure it's in the range of io? Don't want to look for later versions that can't be found.

**Comment by stephenmichaelf** [comment]
Let's add tests for this. And make sure it doesn't try to find more recent versions that can't be found. E.g. - only look for lower versions that we know come from io.

**Comment by damccorm** [comment]
Looks like node is actually no longer allowing you to do this sort of download. Previously, if I went to https://iojs.org/download/release/v3.3.1/ I could download iojs, but that is no longer available (though it still shows up [here](https://nodejs.org/en/download/releases/), the links just don't work). Unclear whether this is intentional or not, but I tend to think if they've dropped support to the point where this is no longer noticed we probably shouldn't be supporting this either.

@bryanmacfarlane any thoughts on this? I'm inclined to close here and just not support iojs as part of this tool - it was always unclear whether that was something we should be doing anyways.

**Comment by damccorm** [comment]
To be clear, if they aren't supporting those links anymore I'm not sure we _**can**_ do this

**Comment by stephenmichaelf** [comment]
It makes sense to me to just not support it and close this.

**Comment by damccorm** [comment]
Agreed, I'll close here and in the issue.

**Pull request closed without merging**